### ISSUE TITLE
Some question about ms-ssim loss

### ISSUE THREAD

**Issue by Spacei567** [psyrocloud/MS-SSIM_L1_LOSS#4]
Hi, your implementation is amazing!

I have some question about the convergence of ms-ssim loss. I trained a network but the convergence of ms-ssim loss was so bad, can you give me some advice?

**Comment by psyrocloud** [comment]
> Hi, your implementation is amazing!
> 
> I have some question about the convergence of ms-ssim loss. I trained a network but the convergence of ms-ssim loss was so bad, can you give me some advice?

Thank you for your comment. Please try basic L1 or L2 loss on your network first. If the training converges with L1 or L2 loss, the MS-SSIM loss would work theoretically. If you have trouble with defocus dataset, please try our paper's defocus dataset first. The reference is "Li, Yunpeng, et al. "Eliminating unbalanced defocus blur with a binocular linkage network." Applied Optics 60.5 (2021): 1171-1181." and the URL is https://opg.optica.org/ao/abstract.cfm?uri=ao-60-5-1171

**Comment by Spacei567** [comment]
Hi, Thanks you reply!

In your code the sigma = 0.5, 1.0, 2.0, 4.0, 8.0,and the window_size = 33,  but the other implementation of ms-ssim loss the sigma was sat as 1.5 and window_size = 11 .  when I trained the CNN for Image Restoration, the convergence of ms-ssim loss 
 in other's implementation of ms-ssim loss sat as 1.5 and window_size 11 is so bad, but when I use your code, the  convergence of ms-ssim loss be better, can you tell me why ? I really want to konw the reason.  And I want to konw the difference of your code sigma = 0.5, 1.0, 2.0, 4.0, 8.0,and the window_size = 33 and other implenmentation sigma=1.5 and window_size = 11, can you help me?

**Comment by psyrocloud** [comment]
> Hi, Thanks you reply!
> 
> In your code the sigma = 0.5, 1.0, 2.0, 4.0, 8.0,and the window_size = 33, but the other implementation of ms-ssim loss the sigma was sat as 1.5 and window_size = 11 . when I trained the CNN for Image Restoration, the convergence of ms-ssim loss in other's implementation of ms-ssim loss sat as 1.5 and window_size 11 is so bad, but when I use your code, the convergence of ms-ssim loss be better, can you tell me why ? I really want to konw the reason. And I want to konw the difference of your code sigma = 0.5, 1.0, 2.0, 4.0, 8.0,and the window_size = 33 and other implenmentation sigma=1.5 and window_size = 11, can you help me?

Hi, in my opinion, the larger the window_size is, the bigger the receptive field of your network catches. The different sigmas combing with the same window_size could provide multi-resolution features for the network. The implementation details are the same as original paper's setup, and the references are as follows:
[1] H. Zhao, O. Gallo, I. Frosio and J. Kautz, "Loss Functions for Image Restoration With Neural Networks," in IEEE Transactions on Computational Imaging, vol. 3, no. 1, pp. 47-57, March 2017
[2] https://github.com/NVlabs/PL4NN
[3] https://github.com/VainF/pytorch-msssim

**Comment by Spacei567** [comment]
Hi, I'm really thank you so much . Thanks your patient explain. Thank you so much again!

**Comment by psyrocloud** [comment]
> Hi, I'm really thank you so much . Thanks your patient explain. Thank you so much again!

: )
